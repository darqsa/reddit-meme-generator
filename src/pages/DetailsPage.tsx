import Details from "../components/Details/Details";
import { IMeme } from "../interfaces/interfaces";

const DetailsPage = (): JSX.Element => {
  const meme: IMeme = {
    title: "hola este coso funciona",
    author: "dios, si es verdad que funciona",
    url: "https://pbs.twimg.com/media/FZPh6S_UsAIez3_?format=jpg&name=medium",
    postLink: "no me puedo creer que este funcionando",
    likes: 99999999,
    subreddit: "LOLOLOLOLOLO",
  };
  return <Details meme={meme} />;
};

export default DetailsPage;
