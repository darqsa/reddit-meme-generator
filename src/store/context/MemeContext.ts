import { createContext } from "react";
import { IMemeContext } from "../../interfaces/interfaces";

const MemeContext = createContext<IMemeContext>({
  memes: [],
});

export default MemeContext;
