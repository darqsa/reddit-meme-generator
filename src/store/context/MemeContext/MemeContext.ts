import { createContext } from "react";
import { IMemeContext } from "../../../interfaces/interfaces";

const MemeContext = createContext<IMemeContext>({
  memes: [],
  dispatch: () => {},
});

export default MemeContext;
