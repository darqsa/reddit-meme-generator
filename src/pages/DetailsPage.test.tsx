import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import MemeContext from "../store/context/MemeContext";

import DetailsPage from "./DetailsPage";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({ id: "18954" }),
}));

describe("Given a DeatailsPage component", () => {
  describe("When instantitated", () => {
    test("Then it should render the meme details", () => {
      const testMemeContext = {
        memes: [
          {
            url: "www.test.com",
            title: "test",
            author: "Tamarindos",
            likes: 5,
            postLink: "www.test/postlink.com",
            subreddit: "subreddit",
            isFavorite: false,
            id: "18954",
          },
        ],
        dispatch: () => {
          jest.fn();
        },
        favoriteMemes: [],
        favoriteMemeDispatch: () => {
          jest.fn();
        },
      };

      render(
        <BrowserRouter>
          <MemeContext.Provider
            value={{
              memes: testMemeContext.memes,
              dispatch: testMemeContext.dispatch,
              favoriteMemes: testMemeContext.favoriteMemes,
              favoriteMemeDispatch: testMemeContext.favoriteMemeDispatch,
            }}
          >
            <DetailsPage />
          </MemeContext.Provider>
        </BrowserRouter>
      );

      const detailsElement = screen.getByText(testMemeContext.memes[0].author);

      expect(detailsElement).toBeInTheDocument();
    });
  });
});
