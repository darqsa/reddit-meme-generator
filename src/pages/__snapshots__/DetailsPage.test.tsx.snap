// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a DeatailsPage component When instantitated Then it should show the details page on the screen 1`] = `
Array [
  <div
    className="header-container"
  >
    <header
      className="sc-jqUVSM bhEHzn header-container__title"
    >
      <span
        className="sc-ftvSup kOocOZ"
      >
        MEME
      </span>
      DETAILS
    </header>
    <div
      className="sc-kDDrLX Kwrti header-container__section-container"
    >
      <a
        href="/home"
        onClick={[Function]}
      >
        <button
          className="sc-gsnTZi dtzzMN"
        >
          Main Page
        </button>
      </a>
      <a
        href="/favorites"
        onClick={[Function]}
      >
        <button
          className="sc-gsnTZi dtzzMN"
        >
          My Favorites
        </button>
      </a>
    </div>
  </div>,
  <div
    className="sc-bczRLJ fGamCS details"
  >
    <div
      className="details__image-container"
    >
      <img
        alt="test"
        className="meme-img"
        src="http://www.test.com"
      />
    </div>
    <section
      className="details__info"
    >
      <span
        className="info__item"
      >
        <strong>
          Title:
        </strong>
          test
      </span>
      <span
        className="info__item"
      >
        <strong>
          Author:
        </strong>
        <a
          href="https://www.reddit.com/user/Tamarindos"
          rel="noreferrer"
          target="_blank"
        >
            Tamarindos
        </a>
      </span>
      <span
        className="info__item"
      >
        <strong>
          Url: 
        </strong>
        <a
          href="www.test/postlink.com"
          rel="noreferrer"
          target="_blank"
        >
            /postlink.com
        </a>
      </span>
      <span
        className="info__item"
      >
        <strong>
          Likes: 
        </strong>
          5
      </span>
      <span
        className="info__item"
      >
        <strong>
          Subreddit: 
        </strong>
        <a
          href="https://reddit.com/subreddit"
          rel="noreferrer"
          target="_blank"
        >
            subreddit
        </a>
      </span>
    </section>
  </div>,
]
`;
